### PR TITLE
coprocessor: Fix return type for builtin aggregation fuction 'AVG' and 'SUM'

### PR DESCRIPTION
**Pull request by spongedu**
corresponding modification for https://github.com/pingcap/tidb/pull/5292

### CODE REVIEW

File: src/coprocessor/select/aggregate.rs [hunk 1]
```diff
@@ -128,8 +128,16 @@ impl AggrFunc for Sum {
...
             return Ok(());
         }
-        let d = box_try!(res.into_dec());
-        collector.push(Datum::Dec(d));
+        let is_float = match res {
+            Datum::F64(_) => true,
+            _ => false,
+        };
+        if is_float {
+            collector.push(Datum::F64(res.f64()));
```
Comment (AndreMouche) [inline src/coprocessor/select/aggregate.rs]: why not use `push(res)` directly?

File: src/coprocessor/select/aggregate.rs [hunk 2]
```diff
@@ -128,8 +128,16 @@ impl AggrFunc for Sum {
...
+        } else {
+            let d = box_try!(res.into_dec());
+            collector.push(Datum::Dec(d));
+        }
```
Comment (AndreMouche) [inline src/coprocessor/select/aggregate.rs]: Could you use match like
```
match res {
   Datum::Null=>.xxx
   Datum::F64(v)=>xxx
   _=>xxx
}
```

**Comment by AndreMouche** [comment]
Any update?

File: src/coprocessor/select/aggregate.rs (hunk 1 above)
Comment (spongedu) [reply src/coprocessor/select/aggregate.rs]: ok

File: src/coprocessor/select/aggregate.rs (hunk 2 above)
Comment (spongedu) [reply src/coprocessor/select/aggregate.rs]: ok, I'll refine here.

**Comment by spongedu** [comment]
Sorry for delay sometime. PTAL @AndreMouche

**Comment by coocood** [comment]
LGTM

**Pull request merged**